app/models/transcription.rb

```diff
@@ -30,7 +30,26 @@ def remove_files_from_storage
     export_files.map(&:purge)
   end
 
+  def lock(current_user)
+    update!(locked_by: current_user.login, lock_timeout: DateTime.now + 3.hours)
+  end
+
+  def locked?
+    lock_timeout && DateTime.now < lock_timeout
+  end
+
+  # Is transcription locked by a user that is not the current user?
+  def locked_by_different_user?(current_user)
+    locked? && current_user.login != locked_by
+  end
+
+  def fresh?(if_unmodified_since)
```

Comment:
It might be worth avoiding / augmenting this method name as it's used heavily in rails controller land etc. I read it and thought it was an rails method not your own.

Reply:
Very true, I didnt think of that. Will update.